File: challenge_001_concat_reverse/nakana/answer.clj
function concat(lst1, lst2){
    if(empty(lst1)) return lst2;
    return cons(first(lst1), concat(rest(lst1), lst2));
}

function reverse(lst){
    if(empty(lst)) return [];
    return cons(first(lst1), concat(rest(lst1), lst2));
}

File: challenge_001_concat_reverse/shoji-ito/answer.js
function reverse(lst) {
  return reverse_(lst, []);
}
private function reverse_(lst1, lst2) {
  if (empty(lst1)) return lst2;
  return reverse_(rest(lst1), cons(first(lst1), lst2));
}


function concat(lst1, lst2) {
  return concat_(reverse(lst1), lst2);
}

private function concat_(lst1, lst2) {
  if (empty(lst1)) return lst2;
  return concat_(rest(lst1), cons(first(lst1), lst2));
}
